src/page-objects/storefront/PageNotFound.ts

```
@@ -0,0 +1,20 @@
+import type { Page, Locator } from '@playwright/test';
+import type { PageObject } from '../../types/PageObject';
+
+export class PageNotFound implements PageObject {
+    public readonly pageNotFoundImage: Locator;
+    public readonly headline: Locator;
+    public readonly pageNotFoundMessage: Locator;
+    public readonly backToShopButton: Locator;
+
+    constructor(public readonly page: Page) {
+        this.pageNotFoundImage = page.getByAltText('Page not found');
+        this.headline = page.getByRole('heading', { name: 'Page not found' });
+        this.pageNotFoundMessage = page.getByText(`We are sorry, the page you're looking for could not be found. It may no longer exist or may have been moved.`);
+        this.backToShopButton = page.getByRole('link', { name: 'Back to shop' });
+    }
+
+    url() {
+        return 'non-existing-page';
```

Maybe you return an error message with:

`return new Error('Function not implemented, because it is a modal page object').message;`

---

Actually I need to visit this url to verify 404 page. see: https://github.com/shopware/shopware/pull/6593/files